From 91af9b7444697a7a2daee57696b4167f13eac84a Mon Sep 17 00:00:00 2001
From: Anubhav Raina <anubhav.raina@arm.com>
Date: Tue, 14 May 2024 16:12:52 +0100
Subject: [PATCH] generate_test_runner: strip quotes from test args

`auto/generate_test_runner.rb` pastes the test name and args as-is which
generates invalid test runners when one of them is a string.

* `:cmdline_args: true`
* `:use_param_tests: true`
* and at least one test param macro containing a string.

Signed-off-by: Anubhav Raina <anubhav.raina@arm.com>
---
 auto/generate_test_runner.rb | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/auto/generate_test_runner.rb b/auto/generate_test_runner.rb
index 102f6f3..37e427e 100755
--- a/auto/generate_test_runner.rb
+++ b/auto/generate_test_runner.rb
@@ -414,7 +414,7 @@ class UnityTestRunnerGenerator
           output.puts('      UNITY_PRINT_EOL();')
         else
           test[:args].each do |args|
-            output.puts("      UnityPrint(\"  #{test[:test]}(#{args})\");")
+            output.puts("      UnityPrint(\"  #{test[:test]}(#{args.gsub(/(?<!\\)\"/, '')})\");")
             output.puts('      UNITY_PRINT_EOL();')
           end
         end
--
2.25.1

